python/ray/_private/runtime_env/working_dir.py

```diff
@@ -593,33 +404,25 @@ def ensure_runtime_env_setup(pkg_uris: List[str]) -> Optional[str]:
         with FileLock(str(pkg_file) + ".lock"):
             pkg_dir = fetch_package(pkg_uri)
         sys.path.insert(0, str(pkg_dir))
+
     # Right now, multiple pkg_uris are not supported correctly.
     # We return the last one as working directory
     return str(pkg_dir) if pkg_dir else None
 
 
-def override_task_or_actor_runtime_env(
-        runtime_env: Optional[Dict[str, Any]],
-        parent_runtime_env: Dict[str, Any]) -> Dict[str, Any]:
-    if runtime_env:
-        if runtime_env.get("working_dir"):
-            raise NotImplementedError(
-                "Overriding working_dir for actors is not supported. "
-                "Please use ray.init(runtime_env={'working_dir': ...}) "
-                "to configure per-job environment instead.")
-        # NOTE(edoakes): this is sort of hacky, but we manually add the right
-        # working_dir here so the relative path to a requirements.txt file
-        # works. The right solution would be to merge the runtime_env with the
-        # parent runtime env before validation.
-        if parent_runtime_env.get("working_dir"):
-            runtime_env["working_dir"] = parent_runtime_env["working_dir"]
-        runtime_env_dict = RuntimeEnvDict(runtime_env).get_parsed_dict()
-    else:
-        runtime_env_dict = {}
+def setup_working_dir(runtime_env: dict,
+                      context: RuntimeEnvContext,
+                      logger: Optional[logging.Logger] = None):
+    if not runtime_env.get("uris"):
+        return
+
+    global _logger, PKG_DIR
+    prev_logger = _logger
+    prev_pkg_dir = PKG_DIR
+    _logger = logger
+    PKG_DIR = context.session_dir
```

Comment:
Can you please add a brief comment about why this is necessary?

Reply:
Oh yeah, this shouldn't be necessary but we have the hacky PKG_DIR global parametrization -_-

Reply:
If we make the working_dir logic a class instead of a module with global variables we can get rid of it. Let me add a comment